_shared/local_extensions.py

```diff
@@ -158,6 +160,61 @@ def dedent(self, s):
+            self._random_port_numbers = random.shuffle(
+                [
+                    i
+                    for i in range(1025, 10000)  # Ports <=1024 are privileged on Linux.
+                    if i
+                    not in (
+                        # Commonly-used port numbers to avoid.
+                        8080,  # Common alternative HTTP port.
+                        # Port numbers already used by existing Hypothesis apps.
+                        5000,  # h's web process.
+                        5001,  # h's websocket process.
+                        5432,  # h's Postgres service.
+                        5672,  # h's RabbitMQ service.
+                        15672,  # h's RabbitMQ service.
+                        9200,  # h's Elasticsearch service.
+                        3000,  # The Hypothesis client's development server.
+                        3002,  # Alternative port for the client's development server.
+                        3001,  # The client's package server.
+                        8000,  # Bouncer's web process.
+                        9082,  # Via's web process.
+                        9083,  # Via's NGINX process.
+                        3032,  # Via HTML's uWSGI process.
+                        9085,  # Via HTML's NGINX process.
+                        9086,  # Via HTML's NGINX process.
+                        9101,  # Via Test App's web process.
+                        8001,  # LMS's web process.
+                        48001,  # LMS's web-https process.
+                        5433,  # LMS's Postgres service.
+                        5674,  # LMS's RabbitMQ service.
+                        15674,  # LMS's RabbitMQ service.
+                        9099,  # Checkmate's web process.
+                        5434,  # Checkmate's Postgres service.
+                        5673,  # Checkmate's RabbitMQ service.
+                        15673,  # Checkmate's RabbitMQ service.
+                        4000,  # Report's Metabase service.
+                        5435,  # Report's metabase-postgres service.
+                        5436,  # Report's Postgres service.
+                        5437,  # cookiecutter-pypackage-test's Postgres service.
+                        9202,  # cookiecutter-pyapp-test's Elasticsearch service.
+                        9800,  # cookiecutter-pyramid-app-test's web process.
+                        5438,  # cookiecutter-pyramid-app-test's Postgres service.
+                        9201,  # cookiecutter-pyramid-app-test's Elasticsearch service.
+                        5675,  # cookiecutter-pyramid-app-test's RabbitMQ service.
+                        15675,  # cookiecutter-pyramid-app-test's RabbitMQ service.
+                        5050,  # Publisher Account Test Site's web process.
+                        9000,  # BioPub's web process.
+                    )
+                ]
+            )
+
+        return self._random_port_numbers.pop()
```

Comment:
In the future a single run of the cookiecutter may generate more than one random port number, e.g. a web port and a Postgres port, so we don't want it to ever generate the same random number twice within a single run.